[16.0][ADD] crm_partner_required: New module

EmilioPascual opened the pull request:
It makes the field partner required in the opportunity

¿@rafaelbn @yajo @Gelojr @fcvalgar can you review it? Thanks.

MT-10018 @moduon

EmilioPascual committed:
[ADD] crm_partner_required: New module

MT-9298 @moduon

rafaelbn commented:
Thank you very much @EmilioPascual !

I miss the loom video 😉 

Here you are, could you review please?

https://www.loom.com/share/9bf18dfff6564593828d5895a4403a05?sid=7139ad6d-27f1-4f1e-8499-cef9bcd34178

I agree not adding more fields in the kanban view for better UX, and maybe I also agree **if intended** not mandatory the field in the kanban view. This is a question 😄 

With ❤️ , RB.

EmilioPascual commented:
> I agree not adding more fields in the kanban view for better UX, and maybe I also agree **if intended** not mandatory the field in the kanban view. This is a question 😄

I was thinking about it and perhaps the partner should also be mandatory in quick create form. I've pushed new changes. @rafaelbn can you review again?

yajo commented on crm_partner_required/tests/test_crm_partner_required.py:
```diff
@@ -0,0 +1,38 @@
+            opportunity_form.save()
+        opportunity_form.partner_id = self.partner
+        opportunity_form.save()
+        # Check required in opportunity in quick create form
+        opportunity_quick_create_form = Form(
```
issue: IMHO this is a separate unit test. One for each form.

EmilioPascual replied:
Updated. Thanks for comments 🤗

yajo commented on crm_partner_required/tests/test_crm_partner_required.py:
```diff
@@ -0,0 +1,38 @@
+            opportunity_quick_create_form.save()
+        opportunity_quick_create_form.partner_id = self.partner
+        opportunity_quick_create_form.save()
+        # Check required in lead in default form
+        lead_form = Form(self.env["crm.lead"].with_context(default_type="lead"))
+        lead_form.name = "Test Lead"
+        lead_form.save()
```
issue: and this should be a separate test also.

EmilioPascual replied:
Done

yajo reviewed (changes requested):
> the partner should also be mandatory in quick create form

AFAICS this is already fixed, thanks!

rafaelbn reviewed (approved):
👍🏼 OK!

EmilioPascual committed:
[BUILD] crm_partner_required: Set module as rebel

The module adds a required field in a form view, this behavior is incompatible with test of the other modules.

@moduon

rafaelbn commented:
/ocabot merge patch
